src/preprocessor.py

```diff
@@ -97,6 +97,36 @@ def execute(self, var: varlist_util.VarlistEntry,
         pass
 
 
+class PercentTo01Function(PreprocessorFunctionBase):
+    """A PreprocessorFunction which convers the dependent variable's units and values,
+    for the specific case of percentages. ``0-1`` are not defined in the UDUNITS-2
+    library. So, this function handles the case where we have to convert from 
+    ``0-1`` to ``%``.
+    """
+    
+    _std_name_tuple = ('0-1', '%')
+    
+    def execute(self, var, ds, **kwargs):
+        var_unit = getattr(var, "units", "")
+        tv = var.translation #abbreviate
+        tv_unit = getattr(tv, "units", "")
+        if str(tv_unit) == str(var_unit):
+            return ds
+        # 0-1 to %
+        if str(tv_unit) == self._std_name_tuple[0] and str(var_unit) == self._std_name_tuple[1]:
+            ds[tv.name].attrs['units'] = '%'
+            ds[tv.name].values = ds[tv.name].values*100
+            return ds
+        # % to 0-1
+        if str(tv_unit) == self._std_name_tuple[1] and str(var_unit) == self._std_name_tuple[0]:
+            ds[tv.name].attrs['units'] = '0-1'
+            # sometimes % is [0,1] already
```

Comment:
Is this the actual units def in the file? Usually, 0-1 refers to the value range when the units defined as "fraction".

Reply:
For the CM4.5 dataset, the actual units attr is the string '0-1'. This caused some problems, as it's not recognized by cfunits. I'm sure we could ask @jkrasting about the naming scheme for the unit.

Reply:
@jtmims Thanks for clarifying.  I looked at the GFDL field table, and it seems like the variable being read is the ice coverage mask (ice_mask, where 1=ice, and 0=no ice), instead of sea ice concentration/fractional coverage (ice_frac, which ranges from 0 to 1, and would have units of "fraction" or siconc, which ranges from 0--100%). If it is ice_mask, the query is grabbing the wrong variable.

Reply:
Issue #723 has been made to keep note of weird attributes found in GFDL output data